internal/handler/auth_handler_test.go

```diff
@@ -0,0 +1,163 @@
+package handler_test
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golangTroshin/gophermat/internal/handler"
+	"github.com/golangTroshin/gophermat/internal/model"
+)
+
+type MockAuthService struct {
+	RegisterUserFunc     func(login, password string) (*model.User, error)
+	AuthenticateUserFunc func(login, password string) (*model.User, error)
+}
+
+func (m *MockAuthService) RegisterUser(login, password string) (*model.User, error) {
+	if m.RegisterUserFunc != nil {
+		return m.RegisterUserFunc(login, password)
+	}
+	return nil, nil
+}
+
+func (m *MockAuthService) AuthenticateUser(login, password string) (*model.User, error) {
+	if m.AuthenticateUserFunc != nil {
+		return m.AuthenticateUserFunc(login, password)
+	}
+	return nil, nil
+}
+
+func MockSetAuthCookie(userID uint, w http.ResponseWriter) {
+	w.Header().Set("Set-Cookie", "auth_token=mocked_token; HttpOnly")
+}
+
+func TestAuthHandler_Register(t *testing.T) {
+	mockService := &MockAuthService{}
+
+	tests := map[string]struct {
+		requestBody      string
+		mockRegisterFunc func(login, password string) (*model.User, error)
+		expectedStatus   int
+		expectedCookie   bool
+	}{
+		"Valid Registration": {
```

Comment:
Так как Go автоматически заменяет пробелы на подчеркивания в наименованиях тестов, когда выводит результат я бы не рекомендовал использовать пробелы, или рекомендовал бы сразу заменить их на подчеркивания. В противном случае поиск тестов затрудняется.
```suggestion
		"Valid_Registration": {
```